src/MainWindow.cpp

```diff
@@ -2719,3 +2719,36 @@ void MainWindow::openFindReplaceDialog()
         findReplaceDialog->show();
     }
 }
+
+void MainWindow::saveAsView(QString query)
+{
+    // Let the user select a name for the new view and make sure it doesn't already exist
+    QString name;
+    while(true)
+    {
+        name = QInputDialog::getText(this, qApp->applicationName(), tr("Please specify the view name")).trimmed();
+        if(name.isEmpty())
```

Comment:
One minor thing here: technically this should be ```.isNull()``` instead of ```.isEmpty()```. This makes it possible to create views with an empty name. It was ```.isEmpty()``` in the old code and we're not really consistent here but I thought it might make sense to change this while working on this code anyway :smile: